blockmanager.go

```diff
@@ -894,7 +894,7 @@ func (b *blockManager) getCheckpointedCFHeaders(checkpoints []*chainhash.Hash,
 			// Find the first and last height for the blocks
 			// represented by this message.
 			startHeight := checkPointIndex*wire.CFCheckptInterval + 1
-			lastHeight := startHeight + wire.CFCheckptInterval
+			lastHeight := (checkPointIndex + 1) * wire.CFCheckptInterval
```

Comment:
Nice catch! This now properly matches up with the height range generated above. It would seem though, this this should've had larger ramifications? Though I guess we would still generate+write the proper interval, but the check below would at times fail to catch a repeated message. 

Another question: why would we send repeated queries in the first place? Possibly the query peers logic thinks that a requested timeout, but in actuality, the remote peer intended to send it, causing us to issue a double request.

Reply:
It wouldn't have larger ramifications, since it is only used for the boundary check below, and only intervals ending at the last written height (repeated intervals) would pass both the start and end height check.

From inspecting the reported logs, it seems like the repeated intervals would be a result of an initial query timing out, which prompts us to resend the query, receiving two responses. The current query timeout is 3 seconds.